### PR TITLE
Convert configuration values in Alarmcallback Config update

### LINKED ISSUE #1399: Alarm callbacks and inputs need type conversion for configuration fields
These API endpoints can accidentally accept string values for number fields when creating or editing the configuration.

This leads to errors (mostly NPEs) when using the configuration. Use `org.graylog2.utilities.ConfigurationMapConverter#convertValues` to convert to correct values wherever possible.

### PR DESCRIPTION
**Pull request by kroepke**
ensure that alarm callback configurations conform to the declared types in the requested plugin config

 - the web interface had a bug where integers were treated as strings on updates causing the wrong config to be stored
 - also fix an NPE that could happen with exceptions having a null message
 - remove AlarmCallbackConfigurationService#update in favor of load/update object/save to allow for sanitizing the configuration
 - removed various author javadocs which we don't use in our codebase

fixes #1399

### CODE REVIEW

File: graylog2-rest-models/src/main/java/org/graylog2/rest/models/alarmcallbacks/AlarmCallbackError.java [hunk 1]
```diff
@@ -32,6 +33,6 @@
 
     @JsonCreator
     public static AlarmCallbackError create(@JsonProperty("error") String error) {
-        return new AutoValue_AlarmCallbackError(error);
+        return new AutoValue_AlarmCallbackError(Strings.nullToEmpty(error));
```
Comment (joschi) [inline graylog2-rest-models/src/main/java/org/graylog2/rest/models/alarmcallbacks/AlarmCallbackError.java]: Why is this necessary? The error message shouldn't be `null` (or empty for that matter) but mention the reason for the error.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/alarmcallbacks/AlarmCallbackResource.java [hunk 2]
```diff
@@ -207,17 +215,55 @@ public void delete(@ApiParam(name = "streamid", value = "The stream id this alar
...
+                aCC.getCreatorUserId());
+
+        try {
+             alarmCallbackConfigurationService.save(newConfig);
+        } catch (ValidationException e) {
+            throw new BadRequestException("Unable to save alarm callback configuration", e);
+        }
+    }
+
+    private Map<String, Object> convertConfigurationValues(@ApiParam(name = "JSON body", required = true) CreateAlarmCallbackRequest alarmCallbackRequest) {
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/alarmcallbacks/AlarmCallbackResource.java]: Unnecessary `@ApiParam` annotation.

File: graylog2-rest-models/src/main/java/org/graylog2/rest/models/alarmcallbacks/AlarmCallbackError.java (hunk 1 above)
Comment (bernd) [inline graylog2-rest-models/src/main/java/org/graylog2/rest/models/alarmcallbacks/AlarmCallbackError.java]: We ran into this where `error` was null when we had a null pointer exception.

Comment (kroepke) [inline graylog2-rest-models/src/main/java/org/graylog2/rest/models/alarmcallbacks/AlarmCallbackError.java]: This caused an exception when rendering the alarm callback history item, because error is not nullable. Since we didn't want to chase all the usages of this field in the web interface code, we decided to make it "" instead.

**Comment by edmundoa** [comment]
LGTM :+1:
